message_app/static/js/messages/main.js

```diff
@@ -92,10 +95,21 @@ for (const m of messageArrays) {
 const messagePlaceholder = document.getElementsByClassName("message-form__placeholder")[0];
 
 const textEditor = document.getElementsByClassName("message-form__content-editable")[0];
-textEditor.addEventListener("keypress", function(){
-    socket.emit("typing", {"username": username});
-});
+
+// Give Feedback to other users in the room if an user is typing
+// The function will be called every 2 seconds
+setInterval(isTyping, 2000);
+function isTyping(){
+    // An user is typing when their focus is on the textbox and their textbox is not empty
+    if (textEditor.textContent.length > 0 && document.activeElement === textEditor){
+        socket.emit("typing", {"username": username, "is_typing": true});
+    } else {
+        socket.emit("typing", {"username": username, "is_typing": false});
```

Comment:
If the user stop typing, then we can break the interval set in line 101. In order to break an interval, assign it to a variable. Something like this:
```
const isTypingInterval = setInterval(isTyping, 2000);
```
Then, inside the else clause
```
clearInterval(isTypingInterval);
````
With the above mechanism, you have to find a way to see if the textEditor element is clicked and focused. You can do something like this.
```
textEditor.addEventListener("click", => {
    // combine with the logic you have above
});
```